### PR TITLE
Organize and elaborate on flag/option description in convert command help text

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: 

## Summary

This reorganizes the `docc convert --help` output to group command line flags and options by category and adds descriptions to many flags and options to provide additional information on how to use them.

Additionally, this makes the following smaller changes:
- The `preview` command has clarified usage text and a discussion. Unfortunately the organization from the coverage command doesn't cary over. 
- Many long-time-deprecated command line flags and options now print a warning when used.
- The top-level `index` command is hidden (but still functional). It has long been available as a `process-archive` subcommand.
- The documentation coverage flags and options have been renamed ("--level" and "--kinds" was overly general names) and better documented in their help text.
- It's no longer necessary to pass a both "--experimental-documentation-coverage" and "--coverage-summary-level" to enable the default level coverage output. (Previously, passing only "--experimental-documentation-coverage" silently did nothing).

## Dependencies

None

## Testing

- Run `swift run docc convert --help` and inspect the output
- Run `swift run docc preview --help` and inspect the output 
- Run `swift run docc --help` and inspect the output

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~[ ] Added tests~
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
With these changes, the help output for `docc convert --help` looks like this
```
OVERVIEW: Convert documentation markup, assets, and symbol information into a documentation archive.

When building documentation for source code, the 'convert' command is commonly invoked by other tools as part of a build workflow. Such build
workflows can perform tasks to extract symbol graph information and may infer values for 'docc' flags and options from other build configuration.

When building documentation for a catalog that only contain articles or tutorial content, interacting with the 'docc convert' command directly can be
a good alternative to using DocC via a build workflow.

USAGE: docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>]
       docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>]
       docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>] [<availability-options>] [<diagnostic-options>] [<source-repository-options>] [<hosting-options>] [<info-plist-fallbacks>] [<feature-flags>] [<other-options>]

INPUTS & OUTPUTS:
  <docc-catalog-path>     Path to a '.docc' documentation catalog directory.
  --additional-symbol-graph-dir <additional-symbol-graph-dir>
                          A path to a directory of additional symbol graph files.
  -o, --output-path, --output-dir <output-path>
                          The location where the documentation compiler writes the built documentation.

AVAILABILITY OPTIONS:
  --platform <platform>   Specify information about the current release of a platform.
        Each platform's information is specified via separate "--platform" values using the following format: "name={platform name},version={semantic
        version}".
        Optionally, the platform information can include a 'beta={true|false}' component. If no beta information is provided, the platform is
        considered not in beta.

DIAGNOSTIC OPTIONS:
  --warnings-as-errors    Treat warnings as errors
  --ide-console-output, --emit-fixits
                          Format output to the console intended for an IDE or other tool to parse.
  --diagnostics-file, --diagnostics-output-path <diagnostics-file>
                          The location where the documentation compiler writes the diagnostics file.
        Specifying a diagnostic file path implies '--ide-console-output'.
  --diagnostic-filter, --diagnostic-level <diagnostic-filter>
                          Filter diagnostics with a lower severity than this level.
        This option is ignored if `--analyze` is passed.

        This filter level is inclusive. If a level of 'note' is specified, diagnostics with a severity up to and including 'note' will be printed.
        The supported diagnostic filter levels are:
         - error
         - warning
         - note, info, information
         - hint, notice
  --analyze               Include 'note'/'information' level diagnostics in addition to warnings and errors.

SOURCE REPOSITORY OPTIONS:
  --checkout-path <checkout-path>
                          The root path on disk of the repository's checkout.
  --source-service <source-service>
                          The source code service used to host the project's sources.
        Required when using '--source-service-base-url'. Supported values are 'github', 'gitlab', and 'bitbucket'.
  --source-service-base-url <source-service-base-url>
                          The base URL where the source service hosts the project's sources.
        Required when using '--source-service'. For example, 'https://github.com/my-org/my-repo/blob/main'.

HOSTING OPTIONS:
  --transform-for-static-hosting/--no-transform-for-static-hosting
                          Produce a DocC archive that supports static hosting environments. (default: --transform-for-static-hosting)
  --hosting-base-path <hosting-base-path>
                          The base path your documentation website will be hosted at.
        For example, if you deploy your site to 'example.com/my_name/my_project/documentation' instead of 'example.com/documentation', pass
        '/my_name/my_project' as the base path.

INFO.PLIST FALLBACKS:
  --fallback-display-name, --display-name <fallback-display-name>
                          A fallback display name if no value is provided in the documentation catalogs's Info.plist file.
        If no display name is provided in the catalogs's Info.plist file or via the '--fallback-display-name' option, DocC will infer a display name
        from the documentation catalog base name or from the module name from the symbol graph files provided via the '--additional-symbol-graph-dir'
        option.
  --fallback-bundle-identifier, --bundle-identifier <fallback-bundle-identifier>
                          A fallback bundle identifier if no value is provided in the documentation catalogs's Info.plist file.
        If no bundle identifier is provided in the catalogs's Info.plist file or via the '--fallback-bundle-identifier' option, DocC will infer a
        bundle identifier from the display name.
  --default-code-listing-language <default-code-listing-language>
                          A fallback default language for code listings if no value is provided in the documentation catalogs's Info.plist file.
  --fallback-default-module-kind <fallback-default-module-kind>
                          A fallback default module kind if no value is provided in the documentation catalogs's Info.plist file.
        If no module kind is provided in the catalogs's Info.plist file or via the '--fallback-default-module-kind' option, DocC will display the
        module kind as a "Framework".

DOCUMENTATION COVERAGE (EXPERIMENTAL):
  --experimental-documentation-coverage
                          Generate documentation coverage output.
        Detailed documentation coverage information will be written to 'documentation-coverage.json' in the output directory.
  --coverage-summary-level <symbol-kind>
                          The level of documentation coverage information to write on standard out. (default: brief)
        The '--coverage-summary-level' level has no impact on the information in the 'documentation-coverage.json' file.
        The supported coverage summary levels are 'brief' and 'detailed'.
  --coverage-symbol-kind-filter <symbol-kind>
                          Filter documentation coverage to only analyze symbols of the specified symbol kinds.
        Specify a list of symbol kind values to filter the documentation coverage to only those types symbols.
        The supported symbol kind values are: associated-type, class, dictionary, enumeration, enumeration-case, function, global-variable,
        http-request, initializer, instance-method, instance-property, instance-subcript, instance-variable, module, operator, protocol, structure,
        type-alias, type-method, type-property, type-subscript, typedef

FEATURE FLAGS:
  --experimental-enable-custom-templates
                          Allows for custom templates, like `header.html`.
  --enable-inherited-docs Inherit documentation for inherited symbols
  --allow-arbitrary-catalog-directories
                          Experimental: allow catalog directories without the `.docc` extension.
  --emit-digest           Write additional metadata files to the output directory.
  --emit-lmdb-index       Writes an LMDB representation of the navigator index to the output directory.
        A JSON representation of the navigator index is emitted by default.

OPTIONS:
  -h, --help              Show help information.
```

**Comment by d-ronnqvist** [comment]
After talking to some people about this I made some additional refinements to the order of these options:
- Moved the Diagnostic options group below the Source Repository and Hosting options groups
- Ordered the options in the Diagnostic Options group alphabetically
- Moved the default code listing language option to the top of the Info.plist fallback options group.

With those changes the help text now looks like this:
```
OVERVIEW: Convert documentation markup, assets, and symbol information into a documentation archive.

When building documentation for source code, the 'convert' command is commonly invoked by other tools as part of a build workflow. Such
build workflows can perform tasks to extract symbol graph information and may infer values for 'docc' flags and options from other build
configuration.

When building documentation for a catalog that only contain articles or tutorial content, interacting with the 'docc convert' command
directly can be a good alternative to using DocC via a build workflow.

USAGE: docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>]
       docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>]
       docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>] [<availability-options>] [<diagnostic-options>] [<source-repository-options>] [<hosting-options>] [<info-plist-fallbacks>] [<feature-flags>] [<other-options>]

INPUTS & OUTPUTS:
  <docc-catalog-path>     Path to a '.docc' documentation catalog directory.
  --additional-symbol-graph-dir <additional-symbol-graph-dir>
                          A path to a directory of additional symbol graph files.
  -o, --output-path, --output-dir <output-path>
                          The location where the documentation compiler writes the built documentation.

AVAILABILITY OPTIONS:
  --platform <platform>   Specify information about the current release of a platform.
        Each platform's information is specified via separate "--platform" values using the following format: "name={platform
        name},version={semantic version}".
        Optionally, the platform information can include a 'beta={true|false}' component. If no beta information is provided, the platform
        is considered not in beta.

SOURCE REPOSITORY OPTIONS:
  --checkout-path <checkout-path>
                          The root path on disk of the repository's checkout.
  --source-service <source-service>
                          The source code service used to host the project's sources.
        Required when using '--source-service-base-url'. Supported values are 'github', 'gitlab', and 'bitbucket'.
  --source-service-base-url <source-service-base-url>
                          The base URL where the source service hosts the project's sources.
        Required when using '--source-service'. For example, 'https://github.com/my-org/my-repo/blob/main'.

HOSTING OPTIONS:
  --hosting-base-path <hosting-base-path>
                          The base path your documentation website will be hosted at.
        For example, if you deploy your site to 'example.com/my_name/my_project/documentation' instead of 'example.com/documentation', pass
        '/my_name/my_project' as the base path.
  --transform-for-static-hosting/--no-transform-for-static-hosting
                          Produce a DocC archive that supports static hosting environments. (default: --transform-for-static-hosting)

DIAGNOSTIC OPTIONS:
  --analyze               Include 'note'/'information' level diagnostics in addition to warnings and errors.
  --diagnostics-file, --diagnostics-output-path <diagnostics-file>
                          The location where the documentation compiler writes the diagnostics file.
        Specifying a diagnostic file path implies '--ide-console-output'.
  --diagnostic-filter, --diagnostic-level <diagnostic-filter>
                          Filter diagnostics with a lower severity than this level.
        This option is ignored if `--analyze` is passed.

        This filter level is inclusive. If a level of 'note' is specified, diagnostics with a severity up to and including 'note' will be
        printed.
        The supported diagnostic filter levels are:
         - error
         - warning
         - note, info, information
         - hint, notice
  --ide-console-output, --emit-fixits
                          Format output to the console intended for an IDE or other tool to parse.
  --warnings-as-errors    Treat warnings as errors

INFO.PLIST FALLBACKS:
  --default-code-listing-language <default-code-listing-language>
                          A fallback default language for code listings if no value is provided in the documentation catalogs's Info.plist
                          file.
  --fallback-display-name, --display-name <fallback-display-name>
                          A fallback display name if no value is provided in the documentation catalogs's Info.plist file.
        If no display name is provided in the catalogs's Info.plist file or via the '--fallback-display-name' option, DocC will infer a
        display name from the documentation catalog base name or from the module name from the symbol graph files provided via the
        '--additional-symbol-graph-dir' option.
  --fallback-bundle-identifier, --bundle-identifier <fallback-bundle-identifier>
                          A fallback bundle identifier if no value is provided in the documentation catalogs's Info.plist file.
        If no bundle identifier is provided in the catalogs's Info.plist file or via the '--fallback-bundle-identifier' option, DocC will
        infer a bundle identifier from the display name.
  --fallback-default-module-kind <fallback-default-module-kind>
                          A fallback default module kind if no value is provided in the documentation catalogs's Info.plist file.
        If no module kind is provided in the catalogs's Info.plist file or via the '--fallback-default-module-kind' option, DocC will
        display the module kind as a "Framework".

DOCUMENTATION COVERAGE (EXPERIMENTAL):
  --experimental-documentation-coverage
                          Generate documentation coverage output.
        Detailed documentation coverage information will be written to 'documentation-coverage.json' in the output directory.
  --coverage-summary-level <symbol-kind>
                          The level of documentation coverage information to write on standard out. (default: brief)
        The '--coverage-summary-level' level has no impact on the information in the 'documentation-coverage.json' file.
        The supported coverage summary levels are 'brief' and 'detailed'.
  --coverage-symbol-kind-filter <symbol-kind>
                          Filter documentation coverage to only analyze symbols of the specified symbol kinds.
        Specify a list of symbol kind values to filter the documentation coverage to only those types symbols.
        The supported symbol kind values are: associated-type, class, dictionary, enumeration, enumeration-case, function, global-variable,
        http-request, initializer, instance-method, instance-property, instance-subcript, instance-variable, module, operator, protocol,
        structure, type-alias, type-method, type-property, type-subscript, typedef

FEATURE FLAGS:
  --experimental-enable-custom-templates
                          Allows for custom templates, like `header.html`.
  --enable-inherited-docs Inherit documentation for inherited symbols
  --allow-arbitrary-catalog-directories
                          Experimental: allow catalog directories without the `.docc` extension.
  --emit-digest           Write additional metadata files to the output directory.
  --emit-lmdb-index       Writes an LMDB representation of the navigator index to the output directory.
        A JSON representation of the navigator index is emitted by default.

OPTIONS:
  -h, --help              Show help information.
```

File: Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/Convert.swift [hunk 1]
```diff
@@ -13,331 +13,578 @@ import SwiftDocC
...
-
-        /// The user-provided path to a `.docc` documentation bundle.
-        @OptionGroup()
-        public var documentationBundle: DocumentationBundleOption
-
+            abstract: "Convert documentation markup, assets, and symbol information into a documentation archive.",
+            usage: """
+            docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>]
+            docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>]
+            docc convert [<catalog-path>] [--additional-symbol-graph-dir <symbol-graph-dir>] [--output-dir <output-dir>] [<availability-options>] [<diagnostic-options>] [<source-repository-options>] [<hosting-options>] [<info-plist-fallbacks>] [<feature-flags>] [<other-options>]
```
Comment (sofiaromorales) [inline Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/Convert.swift:33]: Minor: Change `<catalog-path>`to `<docc-catalog-path>` to make it match to the INPUTS & OUTPUTS argument name

**Comment by sofiaromorales** [comment]
I suggest listing only `--port-number` and `<docc-catalog-path>` as arguments to the `docc preview` command to improve clarity, otherwise it looks pretty good.

**Comment by d-ronnqvist** [comment]
> I suggest listing only `--port-number` and `<docc-catalog-path>` as arguments to the `docc preview` command to improve clarity, otherwise it looks pretty good.

I haven't found a way to customize the help text for the preview command without repeating the entire convert command and all its options inline. Setting the nested convert command's visibility to hidden or private doesn't hide it from the help text and I don't see any way to customize the command help text as a whole.

The only things I seem to be able to do is to move the port argument above the other unordered arguments.

**Comment by d-ronnqvist** [comment]
I created a small reproducer and filed https://github.com/apple/swift-argument-parser/issues/585 about the issue with the organization of the options for the `preview` command.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Review by sofiaromorales — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**